src/new_index/schema.rs

```diff
@@ -145,6 +145,29 @@ pub struct ScriptStats {
     pub spent_txo_sum: u64,
 }
 
+impl ScriptStats {
+    #[cfg(feature = "liquid")]
+    fn is_sane(&self) -> bool {
+        // See below for comments.
+        self.spent_txo_count <= self.funded_txo_count
+            && self.tx_count <= self.spent_txo_count + self.funded_txo_count
+    }
+    #[cfg(not(feature = "liquid"))]
+    fn is_sane(&self) -> bool {
+        // There are less or equal spends to funds
+        self.spent_txo_count <= self.funded_txo_count
+        // There are less or equal transactions to total spent+funded txo counts
+        // (Most spread out txo case = N funds in 1 tx each + M spends in 1 tx each = N + M txes)
+        && self.tx_count <= self.spent_txo_count + self.funded_txo_count
+        // There are less or equal spent coins to funded coins
+        && self.spent_txo_sum <= self.funded_txo_sum
+        // If funded and spent txos are equal (0 balance)
+        // Then funded and spent coins must be equal (0 balance)
+        && (self.funded_txo_count == self.spent_txo_count)
+            == (self.funded_txo_sum == self.spent_txo_sum)
```

Comment:
This prevents positive balance when there is an equal number of fund and spend txos.

ie. The balance should be 0 but some the balance value is some small amount like a few satoshis.